Feature 269 export csv in chunks

chrwm opened the pull request:
Closes #269 

The feature will allow to export csv's in real chunks and hence enable csv-export on machines with little memory.

Commit:
Add stream_results=True to sqlite engine #269

The option will enable real batching and allow to only load rows from the database when the user requests them. This feature was introduced to allow the data export in chunks without loading all data into the memory. SQLAlchemy is using a client-side cursor: Without stream_results=True it loads all the data into memory, and then hands the Pandas API x-chunksize rows at a time. This could still resolve in memory issue if the data set is big enough - thus, this implementation.

Commit:
Query DB in chunks #269

DB is queried in chunks. Each chunk is checked for duplicate columns and a string correction in two columns, handed over to export function. Applied black.

Commit:
Write csv's in chunks #269

Chunks are written to csv piecewise.

Commit:
Apply black #269

Commit:
Update changelog #269

Commit:
Revert "Add stream_results=True to sqlite engine #269"

This reverts commit f85d1a0829eab87cbee6d59233574514c54f959a.

chrwm commented:
I'll address the refactoring of the csv export functions in a different issue.
Feel free to merge when you approve.

deniztepe committed:
Merge branch 'develop' into feature-269-memory-csv-export

deniztepe commented on open_mastr/soap_api/download.py:
```diff
@@ -395,7 +395,22 @@ def to_csv(df, technology):
+    if os.path.exists(csv_file):
+        df.to_csv(
+            csv_file,
+            mode="a",
```
The new method with 'append' leads to error, if the user tries to export the same tech for the second time, since the table already exists. The csv tables must be first deleted. Is the use case in this manner is intended?

chrwm replied:
I have not considered this to be a use case. In my view, the deletion of tables is a good way to avoid having exported csv tables from different database versions.

FlorianK13 committed:
Apply black #269

FlorianK13 committed:
Introduce chunksize in test #269

FlorianK13 committed:
Write csv according to chunk_number #269

By doing so, the to_csv function also works even if a file
with the same name already existed in the folder. This file is then
deleted before writing the new csv.

FlorianK13 committed:
Chunksize as function parameter #269

FlorianK13 committed:
Chunksize as function parameter #269

FlorianK13 committed:
Merge branch 'develop' into feature-269-memory-csv-export

FlorianK13 committed:
Correct data type of PLZ #269

Should also solve error from #319

FlorianK13 committed:
Test API download with random technologies #269

To decrease the time used for unit tests

FlorianK13 committed:
Delete usage of api_processes #269

Also part of #274

deniztepe committed:
Speed up reverse backfill basic units by filling only required techs #269

deniztepe committed:
Merge remote-tracking branch 'origin/develop' into feature-269-memory-csv-export

FlorianK13 committed:
Delete usage of data_io from csv test #269

deniztepe committed:
Add csv header only once #269

deniztepe committed:
Try to correct test_to_csv #269

Path to csv is erroneous on github ci
